### PR TITLE
Add RadioCardButton component

### LINKED ISSUE #1273: Add radio card button component
image

### PR DESCRIPTION
**Pull request by drillprop**
fix #1273 
design: https://www.figma.com/file/mMGcjK0usOHzKXAwcBdwiv/[-10]-Upload-flow-improvements?node-id=3%3A7236

### CODE REVIEW

**Comment by kdembler** [comment]
@KubaMikolajczyk shouldn't this have some kind of hover state specified? I don't see anything in the design

File: src/views/studio/EditVideoSheet/EditVideoForm/EditVideoForm.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ import { Datepicker } from '@/shared/components/Datepicker'
 import { FormField } from '@/shared/components/FormField'
 import { FileErrorType, ImageInputFile, VideoInputFile } from '@/shared/components/MultiFileSelect/MultiFileSelect'
```
Comment (kdembler) [inline src/views/studio/EditVideoSheet/EditVideoForm/EditVideoForm.tsx]: 😅 

```suggestion
import { FileErrorType, ImageInputFile, VideoInputFile } from '@/shared/components/MultiFileSelect'
```

File: src/views/studio/EditVideoSheet/EditVideoForm/EditVideoForm.tsx [hunk 2]
```diff
@@ -487,25 +480,40 @@ export const EditVideoForm: React.FC<EditVideoFormProps> = ({
...
+                        onChange={() => onChange(true)}
+                        selectedValue={value?.toString()}
+                        helperText="Visible to all"
+                      />
+                      <RadioCardButton
+                        value="false"
+                        label="Unlisted"
+                        onChange={() => onChange(false)}
+                        selectedValue={value?.toString()}
+                        helperText={'Visible with link only'}
```
Comment (kdembler) [inline src/views/studio/EditVideoSheet/EditVideoForm/EditVideoForm.tsx]: Let's use regular `"` here

File: src/views/studio/EditVideoSheet/EditVideoForm/EditVideoForm.tsx [hunk 3]
```diff
@@ -487,25 +480,40 @@ export const EditVideoForm: React.FC<EditVideoFormProps> = ({
...
+                  name="isPublic"
+                  control={control}
+                  defaultValue={true}
+                  rules={{
+                    validate: (value) => value !== null,
+                  }}
+                  render={({ field: { value, onChange, ref } }) => (
+                    <RadioCardButtonsContainer>
+                      <RadioCardButton
+                        ref={ref}
```
Comment (kdembler) [inline src/views/studio/EditVideoSheet/EditVideoForm/EditVideoForm.tsx]: Why do we pass the `ref` here?

File: src/shared/components/RadioButton/RadioButton.style.ts [hunk 4]
```diff
@@ -16,70 +15,6 @@ type RadioButtonCaptionProps = {
...
-          background-color: ${colors.transparentPrimary[6]};
-        }
-      }
-    `
-  }
-}
-
 export const RadioButtonLabel = styled.label<RadioButtonStyleProps>`
   --radio-button-size: 18px;
   --radio-button-container-size: ${sizes(6)};
```
Comment (kdembler) [inline src/shared/components/RadioButton/RadioButton.style.ts]: `--radio-button-size` isn't used anymore, let's remove it

`--radio-button-container-size` is used for `RadioButtonContainer` but I don't think we really need it? Tbh we shouldn't need `RadioButtonContainer` at all

File: src/shared/components/RadioButton/RadioButton.tsx [hunk 5]
```diff
@@ -1,43 +1,38 @@
...
-  Input,
   RadioButtonContainer,
   RadioButtonLabel,
   RadioButtonStyleProps,
   StyledHelperText,
-  StyledInput,
   StyledLabelText,
 } from './RadioButton.style'
 
+import { RadioInput } from '../RadioInput/RadioInput'
```
Comment (kdembler) [inline src/shared/components/RadioButton/RadioButton.tsx]:
```suggestion
import { RadioInput } from '../RadioInput'
```

File: src/shared/components/RadioCardButton/RadioCardButton.styles.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  }
+  return colors.gray[500]
+}
+
+export const RadioCardLabel = styled.label<RadioCardLabelProps>`
+  padding: ${sizes(4)};
+  display: flex;
+  flex-direction: column;
+  cursor: pointer;
+  border: 1px solid ${({ selected, disabled, error }) => getRadioCardBorderColor({ selected, disabled, error })};
```
Comment (kdembler) [inline src/shared/components/RadioCardButton/RadioCardButton.styles.ts]: You can just pass the function, it should get called with the props:
```suggestion
  border: 1px solid ${getRadioCardBorderColor};
```

**Review by kdembler — CHANGES_REQUESTED** [review]

**Comment by drillprop** [comment]
> @KubaMikolajczyk shouldn't this have some kind of hover state specified? I don't see anything in the design

Found it: https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=750%3A90714
I'll update this

**Comment by drillprop** [comment]
@kdembler In the design, this component is called `option card`, but I'm not sure if we should add this. I think we could just stick to RadioCardButton for now. I don't see the usage of the checkbox with this card yet. Let me know what you think

**Comment by kdembler** [comment]
I think here it makes sense to use the name from the design - our name clearly works only for radio buttons while the design allows for checkboxes inside as well. I wouldn't want us to need to change the component name later when we add checkboxes

**Comment by KubaMikolajczyk** [comment]
The way those option cards work is flawless - only issues I spotted are in their responsiveness:
image
Somehow they stop filling the container after 865px breakpoint 
image
And on mobile those should use a vertical layout

Everything is explained here: https://www.figma.com/file/xm6vBdSnQDroc57rGfGtPM/Page-Video-workspace?node-id=314%3A6148

**Review by kdembler — COMMENTED** [review]
All looks good but I think we still have a bit of inconsistency with the hover state. Design says that on card hover, the radio button should also have its hover state triggered. However it seems to me this may be complicated. Do you have any idea on how to achieve that?

**Comment by drillprop** [comment]
> All looks good but I think we still have a bit of inconsistency with the hover state. Design says that on card hover, the radio button should also have its hover state triggered. However it seems to me this may be complicated. Do you have any idea on how to achieve that?

Check now.
I think it's relatively clean. I mean I kinda don't like that we need to import `CustomRadioInput` in `OptionCard`, but didn't find a cleaner solution.

**Review by kdembler — APPROVED** [review]

**Pull request merged**